### ISSUE TITLE
Querying for non-single results is broken

### ISSUE THREAD

**Issue by caseycrogers** [supabase/supabase-flutter#430]
**Describe the bug**
You get the following exception if you try to query for result sets less than or greater than 1 (it's as if all queries were silently being treated as `single` queries):

```Error: PostgrestException(message: JSON object requested, multiple (or no) rows returned, code: PGRST116, details: Results contain 4 rows, application/vnd.pgrst.object+json requires 1 row, hint: null)```

**To Reproduce**
1. Query an arbitrary table with either 0 or greater than 1 rows:
```
await supabase
        .from('foo')
        .select<dynamic>()```
2. Observe `...multiple (or no) rows returned....` exception

**Expected behavior**
I expect to get the query results with no exception.

**Version (please complete the following information):**
On Windows
```
├── supabase_flutter 1.6.2
│   ├── supabase 1.6.2
│   │   ├── functions_client 1.1.1
│   │   ├── gotrue 1.5.7
│   │   ├── postgrest 1.2.3
│   │   ├── realtime_client 1.0.3
│   │   ├── storage_client 1.2.5
```

**Additional context**
Add any other context about the problem here.

**Comment by dshukertjr** [comment]
Thanks for reporting. Looking into this right now.

**Comment by caseycrogers** [comment]
Thanks for the quick response! Looking forward to when this goes live-I've hot patched my local code in the meantime.

**Comment by Vinzent03** [comment]
That should already be fixed in `supabase: ^1.6.3`, so please run `dart pub upgrade supabase`.

**Comment by caseycrogers** [comment]
I'm using `supabase_flutter` and `1.6.3` isn't available on there yet.

**Comment by Vinzent03** [comment]
Yeah, `supabase_flutter` is not `supabase`. `supabase_flutter` is a wrapper around `supabase`. So please update the transitive dependency `supabase` with the command stated previously.

**Comment by caseycrogers** [comment]
Cool done.

You've still got a live bug for anyone who added `supabase_flutter` before `supabase:1.6.3`, why not bump `supabase_flutter`'s pubspec to depend on a non-bugged version of `supabase` so it's obvious to developers that there's a bug fix available?

**Comment by Vinzent03** [comment]
We have three layers of supabase packages. Doing a release in every package above is more work and I think a Flutter/Dart developer should know that every package has itself dependencies. That's why `dart pub outdated` exists. The `supabase-flutter` doesn't depend on the broken version specifically. It specifies a minimum version and that's it. Which exact version is used is set in `pubspec.lock`. When first time installing a package it installs the latest version of all deps. So new user don't have this.

**Comment by caseycrogers** [comment]
Anyone who installed `supabase_flutter` before `supabase:1.6.3` and hasn't run `flutter pub upgrade` after 1.6.3 is currently on a broken version. The conventional way to release a bugfix is to, well, release a version with the bugfix. The first thing I did on identifying this bug is check the `supabase_flutter` changelog to see if there was a bug fix released that I should upgrade to. Relying on developers to discover the bug fix through a transitive dependency is, to be blunt, sloppy. 

If there's a bugged version of `supabase` `supabase_flutter` SHOULD explicitly require a version after the bug was patched. If managing releases is too high friction to publish bug fixes across all your packages then it sounds like you need to better automate your package release process.

Then again, it's your repo not mine so I'll drop it. Just take my (admittedly a bit grumpy) feedback FWIW.

**Comment by Vinzent03** [comment]
I can follow your argumentation and partially agree. I see Tyler and Daniel gave a thumbs up. I **definitely** agree that the release flow needs to be improved. Though I think it's not easy to achieve if we keep the "repo for every package" style. I've proposed a mono repo style a time ago in #199.

@dshukertjr What again was the latest status on #336? I think you've described some issues in some pr.

**Comment by dshukertjr** [comment]
Yup, totally agree. We will try to sort these out the post-launch week.
